src/Assets.php

```diff
@@ -61,17 +61,10 @@ public static function register_assets() {
 	 * Attach data to registered assets using inline scripts.
 	 */
 	public static function maybe_add_asset_data() {
-		if ( wp_script_is( 'wc-shared-settings', 'enqueued' ) ) {
+		if ( wp_script_is( 'wc-vendors', 'enqueued' ) && ! did_action( 'woocommerce_shared_settings' ) ) {
```

Comment:
One potential problem here is that this means anything using shared settings needs to have `wc-vendors` as a dependency right?  I still think we should have a designated "wc-settings" handle used for enqueuing a shared settings bundle (probably `@woocommerce/settings` exported to `wc.sharedSettings` global) so that we have a explicit designated bundle JUST for settings).  Attaching the settings to the `wc-vendors` handle feels fragile because `wc-vendors` doesn't depend on the global.  So it's fairly prone to human error to make sure it's listed as a dependency for something _requiring_ the settings.

Reply:
Also, will this change _require_ the settings package to be published in order to consume?

Reply:
Hmm, which I see appears to have already been published.  I assumed it wasn't published yet based introducing it still being in review.  Should we publish packages (as a stable version) before completed review?

Reply:
I replied to the scold in the other PR. I didn't intend on publishing stable but it happened so ran with it. It can be unpublished if it's a big issue..it's not useful alone anyway, it needs the PHP side implementation to do anything.

> One potential problem here is that this means anything using shared settings needs to have wc-vendors as a dependency right?

Can you help with the webpack config for that? `wc-vendors` contains the components which require the settings, so at least in the case of blocks, as long as the variables are included before that this code works.

Reply:
> Can you help with the webpack config for that?

Essentially it'd be along the lines of what I originally did.  The trickiness mostly relates to adding a shim to wc-admin so it preps it's own fallback in case the `wc-shared-settings` bundle is not available (earlier version of woocommerce without the updated blocks containing the new bundle exposed on the `wc-shared-settings` script).

Want me to take a go at that then?

Reply:
@nerrad Yes please.

Reply:
I'm going to incorporate this as a part of #956 (or derivatives from that).